lib/SecretsCrypto/sf-secrets-interaction-request.c

```diff
@@ -77,29 +282,272 @@ static void _sf_secrets_interaction_request_get_property(GObject *object, guint
+const gchar *sf_secrets_interaction_request_get_secret_name(SfSecretsInteractionRequest *request)
 {
-    (void)request;
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->secret_name;
 }
 
-void sf_secrets_interaction_request_return_error(SfSecretsInteractionRequest *request, const GError *error)
+const gchar *sf_secrets_interaction_request_get_collection_name(SfSecretsInteractionRequest *request)
 {
     SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->collection_name;
+}
+
+const gchar *sf_secrets_interaction_request_get_plugin_name(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->plugin_name;
+}
+
+const gchar *sf_secrets_interaction_request_get_application_id(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->application_id;
+}
 
-    (void)priv;
-    (void)error;
+SfSecretsOperation sf_secrets_interaction_request_get_operation(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->operation;
+}
+
+const gchar *sf_secrets_interaction_request_get_authentication_plugin_name(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->authentication_plugin_name;
+}
+
+const gchar *sf_secrets_interaction_request_get_prompt(SfSecretsInteractionRequest *request,
+        SfSecretsPrompt prompt)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return g_hash_table_lookup(priv->prompt_text, GINT_TO_POINTER(prompt));
+}
+
+GHashTable *sf_secrets_interaction_request_get_prompt_text(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return g_hash_table_ref(priv->prompt_text);
+}
+
+SfSecretsInputType sf_secrets_interaction_request_get_input_type(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->input_type;
+}
+
+SfSecretsEchoMode sf_secrets_interaction_request_get_echo_mode(SfSecretsInteractionRequest *request)
+{
+    SfSecretsInteractionRequestPrivate *priv = sf_secrets_interaction_request_get_instance_private(request);
+    return priv->echo_mode;
+}
+
+static gboolean _sf_hash_table_equal(GHashTable *a, GHashTable *b, GEqualFunc value_equal)
+{
+    GHashTableIter i;
+    gpointer key;
+    gpointer value;
+
+    if (a == b)
```

Comment:
If both are null, then we're returning true which is kind of ok. Just wondering the order of branches and if it this should return false whenever either is null.

Reply:
I think this is quite reasonable as it is; an improvement could be to consider NULL and empty equal, but as code elsewhere tries to avoid creating empty HTs, should be good as-is.